pkg/util/k8sutil/tpr_util.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2016 The etcd-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sutil
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/coreos/etcd-operator/pkg/spec"
+
+	"k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+// UpdateClusterTPRObject updates the given TPR object.
+// ResourceVersion of the object MUST be set or update will fail.
```

Comment:
What does it mean "or update will fail"?
What I am seeing here is that if rv not set then update will always succeed..